### PR TITLE
Comments for brute force and util functions

### PR DESCRIPTION
**Pull request by maluo**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -74,7 +74,7 @@ int main(int argc, char **argv)
                 strcat(bruteforce_characters, "ABCDEFGHIJKLMNOPQRSTUVWXYZ");
                 break;
             case 'l':
-                strcat(bruteforce_characters, "abcdefghijklmnopqrstuvwxyz");
+                strcat(bruteforce_characters, "abcdeJfghijklmnopqrstuvwxyz");
```
Comment (wandersonca) [inline src/main.c]: Can you remove the J

File: src/main.c [hunk 2]
```diff
@@ -118,6 +118,13 @@ int main(int argc, char **argv)
     }
     else
     {
+
+        /*
+        * password_hash - hashed password (with sha256); bruteforce characters - character sets allowed in password;
+        * character_length - by default is 4, but will pass it with -c as arguments as code running;
+        * verbose_flag - debugging message in our program which might not want to show;
+        */
+        result = bruteforce_crack(password_hash, bruteforce_characters, character_length, verbose_flag);
```
Comment (wandersonca) [inline src/main.c]: can you remove this? it is a duplicate of line 116

File: src/bruteforce/bruteforce.h [hunk 3]
```diff
@@ -1,6 +1,9 @@
 #ifndef BRUTEFORCE_H
 #define BRUTEFORCE_H
 
+#define PASS_FOUND 0
```
Comment (wandersonca) [inline src/bruteforce/bruteforce.h]: These are not used let's remove them.
Also I created a global.h for these.

File: src/bruteforce/bruteforce-serial.c [hunk 4]
```diff
@@ -27,7 +41,11 @@ int bruteforce_crack(char *password_hash, char *characters, int password_max_len
             // generate password, hash it, then compare it
             generate_password(i, characters, number_of_characters, j, passwordToTest);
             hash(passwordToTest, buffer);
-            if (!strcmp(password_hash, buffer))
+            /*
+            * We are going to print out @passwordToTest directly if @password_hash aligns with @buffer.
+            */
+            result = findPasswordOrNo(password_hash, buffer, passwordToTest);
+            if (!result)
```
Comment (wandersonca) [inline src/bruteforce/bruteforce-serial.c]: let's switch this back

File: src/bruteforce/bruteforce-omp.c [hunk 5]
```diff
@@ -54,7 +71,7 @@ int bruteforce_crack(char *password_hash, char *characters, int password_max_len
     // Print not found result
     if (result == NOT_FOUND)
     {
-        printf("Password not found.\n");
+        printPassIfFound("",PASS_NOT_FOUND);
```
Comment (wandersonca) [inline src/bruteforce/bruteforce-omp.c]: Let's switch this back.

File: src/bruteforce/bruteforce-mpi.c [hunk 6]
```diff
@@ -53,6 +69,11 @@ int bruteforce_crack(char *password_hash, char *characters, int password_max_len
                     printf("Password found: %s\n", passwordToTest);
                     result = FOUND;
                 }
+                /*
+                * We ahve tried to use the common methods here, but mpi does not support a return statemnt in the middle of logic
+                * An error will be thrown, [Exit code: 1] 
+                */
+                //result = findPasswordOrNo(password_hash, buffer, passwordToTest);
```
Comment (wandersonca) [inline src/bruteforce/bruteforce-mpi.c:76]: Please remove these commented out lines

Comment (maluo) [reply src/bruteforce/bruteforce-mpi.c:76]: Just did this.

**Pull request merged**